internal/component/metrics/type.go

```diff
@@ -22,14 +22,34 @@ type StatTimer interface {
 // StatGauge is a representation of a single gauge metric stat. Interactions
 // with this stat are thread safe.
 type StatGauge interface {
-	// Set sets the value of a gauge metric.
-	Set(value int64)
+	StatGaugeInter
+	StatGaugeFloater
```

Comment:
The same treatment could be done for `StatCounter` (easier, passthrough to gauge) and maybe `StatTimer`

Reply:
I will work on `StatCounter` next